tests/src/builder-tests.ts

```diff
@@ -376,7 +376,38 @@ test('Estimate token transfer fee', async () => {
   const estimateFee = new BigNum(transactionByteLength * estimateFeeRate);
   const resultEstimateFee = await estimateTransfer(transaction);
 
-  expect(fetchMock.mock.calls.length).toEqual(1);
+  fetchMock.mockOnce(`${estimateFeeRate}`);
+  const network = new StacksTestnet();
+  const resultEstimateFee2 = await estimateTransfer(transaction, network);
+
+  expect(fetchMock.mock.calls.length).toEqual(2);
   expect(fetchMock.mock.calls[0][0]).toEqual(apiUrl);
+  expect(fetchMock.mock.calls[1][0]).toEqual(network.transferFeeEstimateApiUrl);
   expect(resultEstimateFee.toNumber()).toEqual(estimateFee.toNumber());
+  expect(resultEstimateFee2.toNumber()).toEqual(estimateFee.toNumber());
+});
+
+test('Transaction broadcast', async () => {
+  const recipient = standardPrincipalCV('SP3FGQ8Z7JY9BWYZ5WM53E0M9NK7WHJF0691NZ159');
+  const amount = new BigNum(12345);
+  const fee = new BigNum(0);
+  const secretKey = 'edf9aee84d9b7abc145504dde6726c64f369d37ee34ded868fabd876c26570bc01';
+  const memo = 'test memo';
+
+  const network = new StacksMainnet();
+
+  const options = {
+    fee,
+    memo: memo,
+  };
+
+  const transaction = await makeSTXTokenTransfer(recipient, amount, secretKey, options);
```

Comment:
I think the general trend in highly parameterized js functions (and one I've noticed at Blockstack), is that we start off with a few reasonable non-optional args and a couple reasonable optional args, then eventually end up with a giant list of top-level optional args. 

I propose we take this breaking change as an opportunity to make all of these building functions take an object (as the only top-level arg), and control the optional and non-optional (now and in the future) args using the cleaner named object param.

Reply:
Yes, I would strongly 👍 this.

Reply:
Great suggestion 👍 We will address this in a separate PR.